.github/workflows/rust.yml

```diff
@@ -165,11 +189,43 @@ jobs:
         run: |
           git diff --staged --no-ext-diff --quiet --exit-code
 
+  publish-release:
+    name: Check version and publish release
+    runs-on: ubuntu-latest
+    needs:
+      ["linux", "mac", "windows"]
+    if: ${{ !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled') }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          merge-multiple: true
+      - name: Publish release if tag doesn't exist
+        id: check-tag
+        if: ${{ github.ref == 'refs/heads/main' && github.event_name == 'push' }}
+        run: |
+          RELEASE_TAG=mozjs-sys-v$(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "mozjs_sys") | .version')
+          git fetch --tags --quiet
+          if ! git show-ref --tags --verify --quiet "refs/tags/${RELEASE_TAG}" ; then
+            gh release create ${{ steps.check-tag.outputs.RELEASE_TAG }} ./*.tar.gz
```

Comment:
`${{ steps.check-tag.outputs.RELEASE_TAG }}` is not available here yet (step is not finished), but `${RELEASE_TAG}` should work. https://github.com/sagudev/mozjs/actions/runs/8213051004/job/22464164401#step:4:5